api/embedded/accelerator.go

```diff
@@ -0,0 +1,192 @@
+package embedded
+
+import (
+	"math/big"
+
+	"github.com/zenon-network/go-zenon/chain/nom"
+	"github.com/zenon-network/go-zenon/common"
+	"github.com/zenon-network/go-zenon/common/types"
+	"github.com/zenon-network/go-zenon/rpc/api"
+	"github.com/zenon-network/go-zenon/rpc/api/embedded"
+	"github.com/zenon-network/go-zenon/vm/constants"
+	"github.com/zenon-network/go-zenon/vm/embedded/definition"
+	"github.com/zenon-wiki/go-zdk/client"
+	"github.com/zenon-wiki/go-zdk/utils/template"
+)
+
+type AcceleratorApi struct {
+	client client.Client
+}
+
+func NewAcceleratorApi(client client.Client) AcceleratorApi {
+	return AcceleratorApi{client}
+}
+
+// RPC
+func (a AcceleratorApi) GetAll(pageIndex, pageSize uint32) (*embedded.ProjectList, error) {
+	if pageSize > api.RpcMaxPageSize {
+		pageSize = api.RpcMaxPageSize
+	}
+	var result embedded.ProjectList
+	err := a.client.Call(&result, "embedded.accelerator.getAll", pageIndex, pageSize)
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+func (a AcceleratorApi) GetProjectById(id types.Hash) (*embedded.Project, error) {
+	var result embedded.Project
+	err := a.client.Call(&result, "embedded.accelerator.getProjectById", id.String())
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+func (a AcceleratorApi) GetPhaseById(id types.Hash) (*embedded.Phase, error) {
+	var result embedded.Phase
+	err := a.client.Call(&result, "embedded.accelerator.getPhaseById", id.String())
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+func (a AcceleratorApi) GetPillarVotes(name string, hashes []types.Hash) ([]definition.PillarVote, error) {
+	var result []definition.PillarVote
+	err := a.client.Call(&result, "embedded.accelerator.getPillarVotes", name, hashes)
+	return result, err
+}
+
+func (a AcceleratorApi) GetVoteBreakdown(id types.Hash) (*definition.VoteBreakdown, error) {
+	var result definition.VoteBreakdown
+	err := a.client.Call(&result, "embedded.accelerator.getVoteBreakdown", id.String())
```

Comment:
I am surprised this doesn't take a context. Usually API calls to external services would accept one.

Reply:
go-zenon's (go-ethereum's) rpc client which mine wraps can take a context
https://github.com/zenon-network/go-zenon/blob/master/rpc/server/client.go
I can expose it on `Call` like I have with subscriptions, but tbh mostly didn't because i wasn't sure what its purpose was